pyuvdata/uvdata.py

```diff
@@ -1575,72 +1575,168 @@ def _key2inds(self, key):
+                     'none': no squeezing of resulting numpy array
+                     'full': squeeze all length 1 dimensions
+            force_copy: Option to explicitly make a copy of the data. Default is False.
+
+        Returns:
+            out: numpy array copy (or if possible, a read-only view) of relevant section of data
+        """
+        force_copy = kwargs.pop('force_copy', False)
+        squeeze = kwargs.pop('squeeze', 'default')
+
+        if len(set(np.ediff1d(indp))) <= 1:
+            p_reg_spaced = True
+            p_start = indp[0]
+            p_stop = indp[-1] + 1
+            if len(indp) == 1:
+                dp = 1
+            else:
+                dp = indp[1] - indp[0]
+        else:
+            p_reg_spaced = False
+
+        if len(ind2) == 0:
+            # only unconjugated baselines
+            if len(set(np.ediff1d(ind1))) <= 1:
+                blt_start = ind1[0]
+                blt_stop = ind1[-1] + 1
+                if len(ind1) == 1:
+                    dblt = 1
+                else:
+                    dblt = ind1[1] - ind1[0]
+                if p_reg_spaced:
+                    out = data[blt_start:blt_stop:dblt, :, :, p_start:p_stop:dp]
+                else:
+                    out = data[blt_start:blt_stop:dblt, :, :, indp]
+            else:
+                out = data[ind1, :, :, :]
+                if p_reg_spaced:
+                    out = out[:, :, :, p_start:p_stop:dp]
+                else:
+                    out = out[:, :, :, indp]
+        elif len(ind1) == 0:
+            # only conjugated baselines
+            if len(set(np.ediff1d(ind2))) <= 1:
+                blt_start = ind2[0]
+                blt_stop = ind2[-1] + 1
+                if len(ind2) == 1:
+                    dblt = 1
+                else:
+                    dblt = ind2[1] - ind2[0]
+                if p_reg_spaced:
+                    out = np.conj(data[blt_start:blt_stop:dblt, :, :, p_start:p_stop:dp])
+                else:
+                    out = np.conj(data[blt_start:blt_stop:dblt, :, :, indp])
+            else:
+                out = data[ind2, :, :, :]
+                if p_reg_spaced:
+                    out = np.conj(out[:, :, :, p_start:p_stop:dp])
+                else:
+                    out = np.conj(out[:, :, :, indp])
+        else:
+            # both conjugated and unconjugated baselines
+            out = np.append(data[ind1, :, :, :], np.conj(data[ind2, :, :, :]), axis=0)
+            if p_reg_spaced:
+                out = out[:, :, :, p_start:p_stop:dp]
+            else:
+                out = out[:, :, :, indp]
+
+        if squeeze is 'full':
+            out = np.squeeze(out)
+        elif squeeze is 'default':
+            if out.shape[3] is 1:
+                # one polarization dimension
+                out = np.squeeze(out, axis=3)
+            if out.shape[1] is 1:
+                # one spw dimension
+                out = np.squeeze(out, axis=1)
```

Comment:
By only checking for squeeze == 'full' or 'defualt', you're treating any other value as equivalent to 'none', which is other acceptable value you listed in the docstring. Is this what you want or should it error if you try to pass it something else?

And are you sure that the no squeeze case should be given by the string 'none' vs the value None?